### DISCUSSION TITLE
ROS dependencies - Validation

### DISCUSSION THREAD

**Discussion by Xindictus** [gstavrinos/ai_msc_robotic_systems_assignments#26] [Q&A]
## System Information

- **Operating System**: Ubuntu 22.04.4 LTS (Jammy Jellyfish)
- **Hardware**: WSL2

## Description

- I've installed the dependencies listed under https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/blob/master/ai_msc.rosinstall
- `rosinstall` didn't work out for me. I used a combination of `vcs` and `rosdep` instead
- `colcon build` is successful with some deprecation warnings (should be ok)

## Validation

Supposing the dependencies were successfully installed, is there a way to easily test they're actually working as expected?

**Comment by gstavrinos** [comment]
Open two terminals:

* On the first one, run `rviz2`

* On the second one, run `ros2 launch tt_umpire sim_tt.launch.py assignment:=1`

After about 30" to 1', everything should be loaded and you should be able to visualize some stuff on rviz, by ticking the equivalent boxes. The screenshot below is a typical outcome.

![rviz_demo](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/6703819/28f1d96f-7f84-4ec6-978b-87d62d5bf9b1)


### Bonus
If everything works, (it most probably won't due to very specific package versions that are required by the assignments), you could also try running the second command with GUI enabled, thus stressing your system even more, but also having a much better outlook of the working environment.

`ros2 launch tt_umpire sim_tt.launch.py assignment:=1 gui:=True`

### Bonus 2

Check out the versions of the packages used in the VM (or even better just copy them via a shared folder) in order to try to replicate the exact setup of the workspace.

**Comment by Xindictus** [comment]
### Wrt `rviz2`

  - Installed `VcXsrv`
  - Used the following exports
    ```
    export XDG_RUNTIME_DIR=/run/user/1000
    export DISPLAY=:0
    export LIBGL_ALWAYS_INDIRECT=0
    ```
  - Sourced workspace's `install/setup.bash`
  - Only ran `rviz2` w/o executing `ros2 launch tt_umpire sim_tt.launch.py assignment:=1` and it produced the following view:

     ![image](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/7975922/d1fc81ea-3786-47d9-b0a9-68880dc11ed1)

  **_Up till this point, does the view look ok?_**

---

### Wrt 2nd CMD

- I'm getting the following error when trying to execute it:
  ```
  [ERROR] [launch]: Caught exception in launch (see debug for traceback): [Errno 2] No such file or directory: '~/xin_ws/install/tiago_2dnav/share/tiago_2dnav/launch/tiago_sim_nav_bringup.launch.py'
  ```

- Tried to check the package versions. The "best" command I found in the docs for that was:
  `colcon list --paths-only`

  However, this returns different results between the WSL2 env and the VM. More specifically:
  
  ![image](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/7975922/c4dc9dd3-6db5-470d-92f2-73167dcf4eb2)
   
  ![image](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/7975922/b8f8c883-1614-4148-9c30-fe0d185d2ae1)

- Also attaching my `.rosinstall` (as `.log` due to `Github` restrictions):
    
  [my.log](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/files/14675750/my.log)

**Comment by gstavrinos** [comment]
* rviz looks ok
* Depending on how you have installed the packages there are various ways to check the version. I would check the tag and commit for packages installed from source and the package.xml version for the rest. package.xml can be used for the packages installed from source too.

Check the recommendations of *Bonus 2* of my first answer for a (slight) chance of installing it in a less painful way.
